designs/initial.md

```diff
@@ -0,0 +1,60 @@
+## `@rest` directive
+
+### Arguments
+
+- route: path to rest endpoint. This could be a path or a full url. If a path, add to the endpoint given on link creation or from the context
+- params: a map of variables to url params
```

Comment:
Can't we use directly the name of the params of the query ?

```graphql
query postTitle($id: ID!) {
  post(id: $id) @rest(type: "Post", endPoint: "/post/:id") {
    id
    title
  }
}
```

In my [experiment](https://github.com/sabativi/apollo-link-rest-api) I haven't found it useful

Reply:
@sabativi sure! That would make it easier.

Also, it looks like you have already done a lot of work on this! Would you have any interest of merging your experiment with this repo and being a maintainer?

Reply:
@jbaxleyiii we can start from my experiment if this can help, I need to integrate `apollo-utilities` first to really have something. I am interested being a maintainer also as I love the subject

Reply:
@sabativi that is SUCH great news! Want to start by moving what you have over? SO cool for maintainer, are you on slack? Can you reach out to me there?

Reply:
Hey, will move my code here tomorrow.
Talk to you on slack then

Reply:
@sabativi I really look forward to working with this! #excited

Reply:
Here it is https://github.com/apollographql/apollo-link-rest/pull/2